lib/travis/guest-api/cache.rb

```diff
@@ -1,57 +1,42 @@
 require 'active_support/core_ext/numeric/time'
+require 'redis'
 
 module Travis::GuestAPI
   # Remembers steps by their jobs so that
   # they can be  provided in route GET steps/:uuid
   # for backward compatibility.
   class Cache
-    def initialize(max_job_time = 24.hours, gc_polling_interval = 1.hour)
-      @cache = {}
+    def initialize(max_job_time = 24.hours)
       @max_job_time = max_job_time
       @mutex = Mutex.new
-      initialize_garbage_collector gc_polling_interval
-    end
-
-    def initialize_garbage_collector(polling_interval)
-      @thread = Thread.new do
-        begin
-          loop do
-            Travis.logger.debug "Next step cache GC in #{polling_interval} seconds."
-            sleep polling_interval.to_i # fails in jruby without to_i
-            gc
-          end
-        rescue StandardError => e
-          Travis.logger.error "Step Cache GC exploded: #{e.class}: #{e.message}"
-          raise e
-        end
-      end
+      @redis = Redis.new
```

Comment:
I propose to add redis = Redis.new(:url => "redis://:p4ssw0rd@10.0.1.1:6380/15") to be redis configurable